pkgs/development/python-modules/beets/default.nix

```diff
@@ -476,6 +477,36 @@ buildPythonPackage rec {
 
             ${beets}/bin/beet -c $out/config.yaml > /dev/null
           '';
+
+      beets-with-beetcamp =
+        (beets.override {
+          pluginOverrides = {
+            beetcamp = {
+              enable = true;
+              propagatedBuildInputs = [ beetcamp ];
+            };
+          };
+        }).overridePythonAttrs
+          {
+            # There's no way to configure this check without disabling it completely.
+            #
+            # It doesn't like that `beetcamp` is built against the previous
+            # `beets` which doesn't include `beetcamp` lol.
+            dontUsePythonCatchConflicts = true;
+          };
```

Comment:
I think the solution to this issue is to modify `beetcamp`'s expression to put `beets` in `nativeBuildInputs` and not `dependencies`. All the beets plugins I touched in https://github.com/NixOS/nixpkgs/pull/452540 had this pattern.